pulsar-tracing/src/main/java/com/datastax/oss/pulsar/tracing/BrokerTracing.java

```diff
@@ -0,0 +1,645 @@
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateProducerDetails(level, producer, traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Producer created: {}", traceDetails);
+  }
+
+  public void producerClosed(ServerCnx cnx, Producer producer, Map<String, String> metadata) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateProducerDetails(level, producer, traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Producer closed: {}", traceDetails);
+  }
+
+  public void consumerCreated(ServerCnx cnx, Consumer consumer, Map<String, String> metadata) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateConsumerDetails(level, consumer, traceDetails);
+    populateSubscriptionDetails(level, consumer.getSubscription(), traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Consumer created: {}", traceDetails);
+  }
+
+  public void consumerClosed(ServerCnx cnx, Consumer consumer, Map<String, String> metadata) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    populateConsumerDetails(level, consumer, traceDetails);
+    populateSubscriptionDetails(level, consumer.getSubscription(), traceDetails);
+
+    traceDetails.put("metadata", metadata);
+
+    log.info("Consumer closed: {}", traceDetails);
+  }
+
+  public void onPulsarCommand(BaseCommand command, ServerCnx cnx) throws InterceptException {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    traceDetails.put("command", command.toString());
+    log.info("Pulsar command called: {}", traceDetails);
+  }
+
+  public void onConnectionClosed(ServerCnx cnx) {
+    if (!getEnabledEvents(cnx).contains(EventReasons.ADMINISTRATIVE)) return;
+
+    TraceLevel level = getTracingLevel(cnx);
+    if (level == TraceLevel.NONE) return;
+
+    Map<String, Object> traceDetails = new TreeMap<>();
+    populateConnectionDetails(level, cnx, traceDetails);
+    log.info("Connection closed: {}", traceDetails);
```

Comment:
I think that we should have a well known and documented list of events and maybe the type of event is simply an entry in the map.
This way they can parse the content with automatic tools and set up alerts.